src/prefect/agent/docker/agent.py

```diff
@@ -442,9 +442,16 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
         # is connected to the default `bridge` network.
         # The rest of the networks are connected after creation.
         if self.networks:
+            network = self.networks[0]
             networking_config = self.docker_client.create_networking_config(
-                {self.networks[0]: self.docker_client.create_endpoint_config()}
+                {network: self.docker_client.create_endpoint_config()}
             )
+
+            # Ensure that we change the network mode from bridge to host/none if using
+            # the special predefined network names
+            if network == "host" or network == "none":
```

Comment:
very nitpicky comment but writing as:

```suggestion
            if network in ("host", "none"):
```

may be a tiny bit better? this is fine though too

Reply:
Sure 🤷